go/webpack/cbor_format_test.go

```diff
@@ -88,3 +114,209 @@ func TestWriteCBOR(t *testing.T) {
+
+	assert.EqualValues(3, d.decodeType(cbor.TypeMap))
+	assert.EqualValues("manifest", d.read(d.decodeType(cbor.TypeText)))
+	manifestStart := d.pos()
+	assert.EqualValues(2, d.decodeType(cbor.TypeMap))
+	// manifest-metadata
+	assert.EqualValues("metadata", d.read(d.decodeType(cbor.TypeText)))
+	assert.EqualValues(2, d.decodeType(cbor.TypeMap))
+
+	assert.EqualValues("date", d.read(d.decodeType(cbor.TypeText)))
+	assert.EqualValues(cbor.TagTime, d.decodeType(cbor.TypeTag))
+	assert.EqualValues(time.Date(2017, 5, 20, 10, 0, 0, 0, time.UTC).Unix(), d.decodeType(cbor.TypePosInt))
+
+	assert.EqualValues("origin", d.read(d.decodeType(cbor.TypeText)))
+
+	assert.EqualValues(cbor.TagURI, d.decodeType(cbor.TypeTag))
+	assert.EqualValues("https://example.com", d.read(d.decodeType(cbor.TypeText)))
+
+	// resource-hashes
+	assert.EqualValues("resource-hashes", d.read(d.decodeType(cbor.TypeText)))
+
+	hashedIndexHtml := concat(
+		cbor.Encoded(cbor.TypeArray, 3),
+		cbor.Encoded(cbor.TypeArray, 8),
+		encodedBytes(":method"), encodedBytes("GET"),
+		encodedBytes(":scheme"), encodedBytes("https"),
+		encodedBytes(":authority"), encodedBytes("example.com"),
+		encodedBytes(":path"), encodedBytes("/index.html?query"),
+		cbor.Encoded(cbor.TypeArray, 6),
+		encodedBytes(":status"), encodedBytes("200"),
+		encodedBytes("content-type"), encodedBytes("text/html"),
+		encodedBytes("expires"), encodedBytes("Mon, 1 Jan 2018 01:00:00 GMT"),
+		// Body:
+		encodedBytes("I am example.com's index.html\n"),
+	)
+
+	assert.EqualValues(2, d.decodeType(cbor.TypeMap))
+	assert.EqualValues("sha256", d.read(d.decodeType(cbor.TypeText)))
+	assert.EqualValues(1, d.decodeType(cbor.TypeArray))
+	sha256IndexHtml := sha256.Sum256(hashedIndexHtml)
+	assert.EqualValues(sha256IndexHtml[:], d.read(d.decodeType(cbor.TypeBytes)))
+
+	assert.EqualValues("sha512", d.read(d.decodeType(cbor.TypeText)))
+	assert.EqualValues(1, d.decodeType(cbor.TypeArray))
+	sha512IndexHtml := sha512.Sum512(hashedIndexHtml)
+	assert.EqualValues(sha512IndexHtml[:], d.read(d.decodeType(cbor.TypeBytes)))
+
+	manifestEnd := d.pos()
+	manifestBytes := cborPack.Bytes()[manifestStart:manifestEnd]
+
+	assert.EqualValues("signatures", d.read(d.decodeType(cbor.TypeText)))
+	assert.EqualValues(1, d.decodeType(cbor.TypeArray))
+	assert.EqualValues(2, d.decodeType(cbor.TypeMap))
+	assert.EqualValues("keyIndex", d.read(d.decodeType(cbor.TypeText)))
+	keyIndex := d.decodeType(cbor.TypePosInt)
+
+	assert.EqualValues("signature", d.read(d.decodeType(cbor.TypeText)))
+	signatureValue := d.read(d.decodeType(cbor.TypeBytes))
+
+	assert.EqualValues("certificates", d.read(d.decodeType(cbor.TypeText)))
+	numCertificates := d.decodeType(cbor.TypeArray)
+	certs := make([]*x509.Certificate, numCertificates)
+	for i := range certs {
+		certBytes := d.read(d.decodeType(cbor.TypeBytes))
+		certs[i], err = x509.ParseCertificate(certBytes)
+		require.NoError(err)
+	}
+	signingCert := certs[keyIndex]
+	signingCert.Verify(x509.VerifyOptions{
```

Comment:
Yes, and I've added an extra test with the wrong DNSName to check that.

And I've added a check that at least one signature covers the package's origin.